vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc

```diff
@@ -203,26 +165,79 @@ void Wallet::GetAnonWalletStatus(ledger::ResultCallback callback) {
 }
 
 void Wallet::DisconnectAllWallets(ledger::ResultCallback callback) {
-  auto wallets = ledger_->ledger_client()->GetExternalWallets();
+  DisconnectWallet(constant::kWalletUphold, callback);
+}
 
-  if (wallets.empty()) {
-    BLOG(1, "No wallets");
-    callback(type::Result::LEDGER_OK);
-    return;
+type::BraveWalletPtr Wallet::GetWallet() {
+  const std::string wallet_string = ledger_->ledger_client()->GetStringState(
+      state::kWalletBrave);
+
+  if (wallet_string.empty()) {
+    return nullptr;
   }
 
-  for (auto& wallet : wallets) {
-    auto wallet_new = ResetWallet(std::move(wallet.second));
-    if (!wallet_new) {
-      continue;
-    }
+  base::Optional<base::Value> value = base::JSONReader::Read(wallet_string);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Parsing of brave wallet failed");
+    return nullptr;
+  }
 
-    ledger_->ledger_client()->SaveExternalWallet(
-        wallet.first,
-        std::move(wallet_new));
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    BLOG(0, "Parsing of brave wallet failed");
+    return nullptr;
   }
 
-  callback(type::Result::LEDGER_OK);
+  auto wallet = ledger::type::BraveWallet::New();
+
+  auto* payment_id = dictionary->FindStringKey("payment_id");
+  if (!payment_id) {
+    return nullptr;
+  }
+  wallet->payment_id = *payment_id;
+
+  auto* seed = dictionary->FindStringKey("recovery_seed");
+  if (!seed) {
+    return nullptr;
+  }
+  std::string decoded_seed;
+  if (!base::Base64Decode(*seed, &decoded_seed)) {
+    BLOG(0, "Problem decoding recovery seed");
+    NOTREACHED();
+    return nullptr;
+  }
+
+  std::vector<uint8_t> vector_seed;
+  vector_seed.assign(decoded_seed.begin(), decoded_seed.end());
+  wallet->recovery_seed = vector_seed;
+
+  return wallet;
+}
+
+void Wallet::SetWallet(type::BraveWalletPtr wallet) {
+  if (!wallet) {
+    BLOG(0, "Brave wallet is null");
+    return;
+  }
+
+  const std::string seed_string = base::Base64Encode(wallet->recovery_seed);
+  std::string event_string;
+  if (wallet->recovery_seed.size() > 1) {
+    event_string = std::to_string(
+        wallet->recovery_seed[0] + wallet->recovery_seed[1]);
```

Comment:
Any reason not to call `SaveEventLog` right here, since that's all we're using `event_string` for?

Reply:
why I do not want to call it inside here is that if something is wrong  with seed, we log empty string so that error is logged

Reply:
Gotcha, makes sense.